### ISSUE TITLE
Option to keep suggestion box "as is" until loading is done?

### ISSUE THREAD

**Issue by magnus-lpa** [AbdulRahmanAlHamali/flutter_typeahead#44]
Maybe I'm doing something wrong, but when testing with suggestions that take only 100 ms to load, the box disappears and reappears, even with 

```
animationStart: 1,
animationDuration: Duration(microseconds: 0),

```
IMHO it would look much nicer if the current suggestions would stay until new ones are loaded. Would it make sense to have an option (or set a duration) for not updating the suggestion box at all until new suggestions are loaded?

**Comment by sjmcdowall** [comment]
Do you have a way to easily recreate this behavior?  k

A little github project based off the stock Flutter project create or something?

**Comment by AbdulRahmanAlHamali** [comment]
That is interesting indeed. So the old suggestions will keep showing until new suggestions have been loaded. I think that might work, but will require some significant rewrite on some parts of the plugin. Also, we will have to make sure that this option plays well with the other options that we currently have, like the animation, etc.
It's definitely worth exploring though.

For now, if you want, we have an option `hideOnLoading` which hides the suggestions box when the suggestions are loading. Would that make it look better for you?

**Comment by wsakka** [comment]
I agree it would be better to maintain the old result set as currently it's quite distracting to the end user (esp if it takes a second or more to return results). At the very least, it should be an option. We may need to uninstall until this behaviour is fixed.

**Comment by sjmcdowall** [comment]
@wsakka -- I have never seen this behavior and we use plugin all over the place.  Can you create (or someone) some reproducible little project that shows this?  

As a work around why can't you await the loading yourself (I assume you're passing in a Future?) and once complete call setState() to rebuild .. or maybe make the callback async and await in there? Can't tell exactly what you are doing .. what is the code?

**Comment by wsakka** [comment]
@sjmcdowall I know you didn't bring it up but my apologies first of all for the tone of my last sentence, upon re-reading it sounded angry in tone when that wasn't my intention. It was just that the project manager did not like the behaviour so we may have to find another solution.

That being said, I captured a video of what I was referring to. Basically, if the user is typing any address (in this case 58 simeon lane), they should have the option of selecting the address that is already there.

https://streamable.com/spsp6

**Comment by sjmcdowall** [comment]
Well that is very interesting!  

I'd like to see the code behind this -- something seems .. odd .. about what you are seeing .. but I certainly agree this isn't a very good User Experience !  

The searching seems pretty slow too .. which doesn't help ..

**Comment by wsakka** [comment]
I was using the Network Link Conditioner to simulate a slow connection (sorry meant to mention that).  I'm away from my computer so I should have some sample code tonight.

**Comment by KaYBlitZ** [comment]
Hi guys. Made some changes in my fork that I believe will address this issue. Haven't found any bugs when I checked but let me know how it works. #66 . My fork to test: https://github.com/KaYBlitZ/flutter_typeahead.git . The changes will make this behavior the default. I believe maintain the old results before the updates is the intended way for this widget to function instead of constantly resizing with the circular progress indicator.

**Comment by KaYBlitZ** [comment]
Make sure to switch to the develop branch when testing....

**Comment by sjmcdowall** [comment]
Yes -- as I learned the hard way ... :) 

YOu need something like:
	git:
		url:   (the URL)
		ref: develop


To get the proper version

**Comment by sjmcdowall** [comment]
Fixed in 1.2.0
